rcl_action/test/rcl_action/test_action_server.cpp

```diff
@@ -28,6 +28,39 @@
 
 #include "test_msgs/action/fibonacci.h"
 
+void * bad_malloc(size_t, void *)
+{
+  return nullptr;
+}
+
+void * bad_realloc(void *, size_t, void *)
+{
+  return nullptr;
+}
+
+/// This is fully defined in action_server.c, reproducing here for testing.
+typedef struct rcl_action_server_impl_t
```

Comment:
This seems brittle since we'll have to remember to update this if the copy in action_server.c changes.
Can we move the definition in action_server.c to a private header that can be included here as well?

Reply:
Moved these structs to an impl directory in include